### DISCUSSION TITLE
Add a button on the `pivot_offset` property in the inspector to center it automatically

### DISCUSSION THREAD

**Discussion by Kartopod** [godotengine/godot-proposals#11281] [Editor]
Very often, you are in a situation where you would want to set the `pivot_offset` of a control node to half it's `size` to move the pivot point to the center of the node. Doing this manually is tedious since you would have to manually type in the values, for *each* axis.

![image](https://github.com/user-attachments/assets/3d6b7be0-a27f-4d49-be0d-a6c708548c0b)

To solve this, I have two ideas:
1. A button that automatically sets the `pivot_offset` to `size/2`.

2. A button to set and lock the pivot offset to the center i.e make it auto update when the size changes, so that you wouldn't need to worry about re-centering the pivot offset when changing the size. 
One consideration with this feature would be whether this behaviour should be applied when the game is playing vs only in the editor.

If either one (Or both!) of these buttons are implemented, the UX would be greatly improved imo.

**Comment by Meorge** [comment]
More customizability for the pivot would be great; I think it's one of the big things I miss from Unity's layout system. I think a good general solution would be allowing a normalized pivot value, such that (0.5, 0.5) would place the pivot at the center, (0, 1) would place it at the bottom right corner, and so on. I recall there being progress being done on this, but I unfortunately don't have the discussions/PRs handy.

**Reply by Mickeon** [reply]
- https://github.com/godotengine/godot/pull/70646
- https://github.com/godotengine/godot/pull/96628 (this has been closed)
